cmd/run.go

```diff
@@ -71,7 +71,9 @@ func bindRunFlags(cmd *cobra.Command, config *app.Config) {
 	cmd.Flags().StringVar(&config.JaegerService, "jaeger-service", "charon", "Service name used for jaeger tracing.")
 	cmd.Flags().BoolVar(&config.SimnetBMock, "simnet-beacon-mock", false, "Enables an internal mock beacon node for running a simnet.")
 	cmd.Flags().BoolVar(&config.SimnetVMock, "simnet-validator-mock", false, "Enables an internal mock validator client when running a simnet. Requires simnet-beacon-mock.")
+	cmd.Flags().StringVar(&config.SimnetValidatorKeys, "simnet-validator-keys", ".charon/validator_keys", "The path to the directory containing simnet validator key shares.")
 	cmd.Flags().BoolVar(&config.BuilderAPI, "builder-api", false, "Enables the builder api. Will only produce builder blocks. Builder API must also be enabled on the validator client. Beacon node must be connected to a builder-relay to access the builder network.")
+	cmd.Flags().StringVar(&config.PrivKeyFile, "private-key", ".charon/charon-enr-private-key", "The path to the enr private key.")
```

Comment:
suggest decoupling flag migration/backwards compatibility from the logic.

```
func bindPrivateKeyFlag(cmd *cobra.Command, privateKeyFile *string) {
  // define `--private-key-file` and `--data-dir` flags here, do overrides and resolving to actual privateKeyFile here. Log or error if anything weird
}
```

Think about edge cases:
 - `datadir` configured and file present, `private-key` configured and not file present
 - `datadir` configred and file not present, `private-key` configured and file present
  - etc etc etc

Reply:
`bindPrivKeyFlag` should only `RESOLVE` to actual private-key-file. It should not handle scenarios in which the keyfile doesn't exist at the `private-key-file` path. This is because all the individual subcommands, ex, `charon create enr`, `charon run`, `charon enr`, `charon bootnode` etc. function differently on the keyfile input. For ex, `charon create enr` would `CREATE` a new file if one doesn't already exist. While, `charon run` will throw an error on startup as it won't create a new one.